sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -231,11 +237,19 @@ public void run() {
         if (flushRequested.get() != null) {
           flush();
         }
-        while (!queue.isEmpty() && batch.size() < maxExportBatchSize) {
+        List<CompletableResultCode> concurrentExports = new ArrayList<>();
+        while (!queue.isEmpty() && concurrentExports.size() < maxConcurrentExports) {
           batch.add(queue.poll().toSpanData());
+          if (batch.size() >= maxExportBatchSize) {
+            concurrentExports.add(exportCurrentBatch());
+          }
+        }
+        if (concurrentExports.isEmpty() && System.nanoTime() >= nextExportTime) {
```

Comment:
what case is this trying to catch? It would only trigger if somehow the queue were empty, but the time had expired...but the queue was just empty! Why would you expect there to be something to export at this point?

Reply:
yeah.. we talked about this last week. I asked @trask to add a comment about it, since it wasn't obvious to me what this was for.